crates/fuel-core/src/schema/block.rs

```diff
@@ -221,19 +223,26 @@ impl BlockQuery {
             (Some(id), None) => id.into(),
             (None, Some(height)) => {
                 let height: u64 = height.into();
-                db.get_block_id(height.try_into()?)?
-                    .ok_or("Block height non-existent")?
+                let data = BlockQueryContext(ctx.data_unchecked());
+                data.block_id(height).map_err(|_| {
+                    async_graphql::Error::new(format!(
+                        "Block with height {} not found",
+                        height
+                    ))
+                })?
             }
             (None, None) => {
                 return Err(async_graphql::Error::new("Missing either id or height"))
             }
         };
 
-        let block = db
-            .storage::<FuelBlocks>()
-            .get(&id)?
-            .map(|b| Block::from(b.into_owned()));
-        Ok(block)
+        let data = BlockQueryContext(ctx.data_unchecked());
+        let block = data.block(id)??;
+
+        Ok(Some(Block {
+            header: crate::schema::block::Header(block.0),
+            transactions: block.1,
+        }))
     }
 
     async fn blocks(
```

Comment:
Do you plan to move functionality below in a separate PR?

Reply:
Yes, was moving over just as a part of a general refactors, but paginated queries will be handled later, they just have some issues with the closure which is taking more time